obj.go

```diff
@@ -150,6 +219,29 @@ func objFromMsg(msg netlink.Message) (Obj, error) {
 		case unix.NFTA_OBJ_TYPE:
 			objectType = ad.Uint32()
 		case unix.NFTA_OBJ_DATA:
+			if !returnLegacyType {
```

Comment:
I think it would be a little cleaner if we reversed control flow here: the legacy type handling could go into a separate function, and this non-legacy code path should be the default (not indented).